students/aminavi/session03/mailroom.py

```diff
@@ -0,0 +1,63 @@
+import sys
+
+
+donor_list = [('William',[ 653772.32 , 953772.32,100000]),('Jeff', [ 8888777.22 ,877.33 ]),('Paul', [663.23, 43.87 ,1.32 ]),('Mark', [1663.23, 4300.87, 10432.00]),('Nigel',[5000000 , 10])]
+
+thank_you_action = "Send a Thank You"
+create_a_report = "Create a Report"
+quit_action = "quit"
+
+def send_thank_you():
+    name = ''
+    while not name or name.lower() == 'list':
+        name = input("Please enter the full name: ")
+        if name.lower() == 'list':
+            print(' '.join(x[0] for x in donor_list))
+
+    donation = int(input("Please enter the donation amount: "))
+    for i in range(len(donor_list)):
+        if name.lower() ==  donor_list[i][0].lower():
+            donor_list[i][1].append(donation)
+            return
```

Comment:
Hmm... If we return here, what about a thank you letter when the donor DB already has an entry for the entered donor name `name`?

I suppose you were trying to solve the case when the entered donor name `name` is not in the donor DB (line 22 case), but it probably should have been fixed by some logic like the following?:
```
    for i in range(len(donor_list)):
        if name.lower() ==  donor_list[i][0].lower():
            donor_list[i][1].append(donation)
            break
    else:
        donor_list.append((name, [donation]))
```
, using Python's unique `else` block in `for`...